frontend/packages/shared/src/contexts/ServicesContext.tsx

```diff
@@ -39,23 +39,19 @@ const handleError = (
 ): void => {
   // TODO : log axios errors
 
-  if (error?.response?.status === ServerCodes.Unauthorized) {
-    const errorMessageKey = 'api_errors.Unauthorized';
-    if (i18n.exists(errorMessageKey)) {
-      toast.error(t(errorMessageKey), { toastId: errorMessageKey, autoClose: LOG_OUT_TIMER_MS });
-    }
+  const errorCode = error?.response?.data?.errorCode;
+  const errorMessageKey = errorCode ? `api_errors.${errorCode}` : 'api_errors.Unauthorized';
+
+  const renderToast = () => {
+    toast.error(t(errorMessageKey), { toastId: errorMessageKey, autoClose: LOG_OUT_TIMER_MS });
     setTimeout(() => {
       logout().then(() => window.location.assign(userLogoutAfterPath()));
     }, LOG_OUT_TIMER_MS);
-    return;
-  }
-
-  const errorCode = error?.response?.data?.errorCode;
-  if (errorCode) {
-    const errorMessageKey = `api_errors.${errorCode}`;
+  };
 
+  if (errorCode || error?.response?.status === ServerCodes.Unauthorized) {
     if (i18n.exists(errorMessageKey)) {
```

Comment:
Can this `if` be moved inside the `renderToast` function?

Reply:
I moved it 😊